src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IDictionaryOfTKeyTValueConverter.cs

```diff
@@ -10,14 +10,14 @@ namespace System.Text.Json.Serialization.Converters
     /// Converter for <cref>System.Collections.Generic.IDictionary{TKey, TValue}</cref> that
     /// (de)serializes as a JSON object with properties representing the dictionary element key and value.
     /// </summary>
-    internal sealed class IDictionaryOfTKeyTValueConverter<TCollection, TKey, TValue>
-        : DictionaryDefaultConverter<TCollection, TKey, TValue>
-        where TCollection : IDictionary<TKey, TValue>
+    internal sealed class IDictionaryOfTKeyTValueConverter<TDictionary, TKey, TValue>
```

Comment:
FWIW from past discussions, "TCollection" was used for all types of collections for ease of code sharing and possible diffing; with this PR reducing the duplicated code, perhaps that is not a concern any longer.